src/pip/_internal/locations/__init__.py

```diff
@@ -64,8 +64,8 @@ def _looks_like_bpo_44860() -> bool:
 
 def _looks_like_red_hat_patched_platlib_purelib(scheme: Dict[str, str]) -> bool:
     platlib = scheme["platlib"]
-    if "/$platlibdir/" in platlib and hasattr(sys, "platlibdir"):
-        platlib = platlib.replace("/$platlibdir/", f"/{sys.platlibdir}/")
+    if "/$platlibdir/" in platlib:
+        platlib = platlib.replace("/$platlibdir/", f"/{_PLATLIBDIR}/")
```

Comment:
This seems unrelated to the rest of the changes.

Reply:
Yup, but it seems like the right thing to do; and I imagine it's here because TP noticed this bug while working on this. :)

Relevant definition has a fallback to `lib` if sys.platlibdir isn't defined.

https://github.com/pypa/pip/pull/10511/files#diff-dcf4783407c2fd65772c5e273a0e0554aa4273dc2242bb32bb41343d135a7498R46

Reply:
Sorry, I wasn't suggesting it shouldn't be done. And I have no objection to it being in this PR. I was mostly flagging it in case there *was* a reason it needed to be here that I'd missed 😉

Reply:
If I remember the details correctly (it’s been a while), this was fixing a MyPy error that somehow only shows up when I was working on this. I don’t know why main isn’t complaining, maybe it’s due to me running Mypy on Windows? Not sure, and this new code is margianlly better anyway.